src/v/cluster/shard_placement_table.cc

```diff
@@ -188,7 +263,14 @@ bytes current_state_kvstore_key(const raft::group_id group) {
 shard_placement_table::shard_placement_table(
   ss::shard_id shard, storage::kvstore& kvstore)
   : _shard(shard)
-  , _kvstore(kvstore) {}
+  , _kvstore(kvstore)
+  , _probe(std::make_unique<probe>()) {
+    if (ss::this_shard_id() == _shard) {
+        _probe->setup_metrics();
+    }
+}
+
+shard_placement_table::~shard_placement_table() = default;
```

Comment:
Why do we need this?

Reply:
Because `probe` definition is not available in the `.h` file.